docs/housing-counselor-data.md

```diff
@@ -0,0 +1,163 @@
+# Housing Counselor Data Processing
+
+The [Find a Housing Counselor tool](https://www.consumerfinance.gov/find-a-housing-counselor/) gets its data from the US Department of Housing and Urban Development. HUD provides an API to their list of approved counseling agencies.
+
+We generate the JSON data for the Find a Housing Counselor tool using the `hud_generate_json` management command. This page describes how that command works to transform HUD data to the JSON file format we use for the FAHC tool, including the following steps:
+  1. fetch agency listings from HUD
+  2. clean the results
+  3. fill in any missing latitude and longitude values
+  4. create files of the 10 nearest results for each US ZIP code
+
+
+## Fetch agency listings
+
+(_in `fetcher.py`_)
+
+Request every housing counselor in HUD's database with this request:
+
+```
+https://data.hud.gov/Housing_Counselor/searchByLocation?Lat=38.8951&Long=-77.0367&Distance=5000
+```
+
+It returns thousands of results like this:
+
+```json
+    {
+        "services": "DFC,FBC,PPC,RHC",
+        "languages": "ENG,SPA",
+        "agc_STATUS": "A",
+        "agc_SRC_CD": "HUD",
+        "counslg_METHOD": "Face to Face Counseling,Group Counseling,Internet Counseling,Phone Counseling",
+        "agcid": "80790",
+        "adr1": "24331 W. Eight Mile Road",
```

Comment:
It occurs to me that it might be a good idea to modify this data (both here and below) into a made-up counselor instead of including specific details of a real one. That way in future if this real one changes its data or is closed, we won't have out-of-date data in our docs. We also wouldn't want to be seen as specifically endorsing or highlighting one counselor.